src/components/IntegrationTests/ScheduledJobsTab.tsx

```diff
@@ -0,0 +1,730 @@
+    const sanitized = sanitizeCronInput(value);
+    const parts = sanitized.trim().split(/\s+/);
+    if (parts.length >= 5) {
+      const isValid = validateSchedule(sanitized);
+      setScheduleError(isValid ? undefined : 'Invalid cron schedule');
+    } else {
+      // Clear error for partial expressions
+      setScheduleError(undefined);
+    }
+  };
+
+  const handleJobNameChange = (value: string) => {
+    setJobName(value);
+    const validation = validateJobName(value);
+    setJobNameError(validation.isValid ? undefined : validation.error);
+  };
+
+  function extractCronJobs(jobs: unknown): CronJob[] {
+    if (!jobs || !Array.isArray(jobs)) {
+      return [];
+    }
+    return jobs.filter((job) => job && job.metadata && job.spec);
+  }
+
+  const loadCronJobs = React.useCallback(async () => {
+    if (!integrationTestsLoaded || integrationTests.length === 0) {
+      return;
+    }
+
+    try {
+      const labelSelector = `application=${applicationName}`;
+      const jobs = await listPeriodicIntegrationTestCronJobs(namespace, labelSelector);
+      const extractedJobs = extractCronJobs(jobs);
+      setCronJobs(extractedJobs);
+    } catch (err) {
+      setError((err as Error).message || 'Failed to load jobs');
+    }
+  }, [namespace, applicationName, integrationTestsLoaded, integrationTests.length]);
+
+  React.useEffect(() => {
+    void loadCronJobs();
+  }, [loadCronJobs]);
+
+  const handleDelete = (name: string) => {
+    setJobToDelete(name);
+    setDeleteModalOpen(true);
+  };
+
+  const closeDeleteModal = () => {
+    setDeleteModalOpen(false);
+    setJobToDelete(null);
+  };
+
+  const confirmDelete = async () => {
+    if (!jobToDelete) return;
+
+    setDeleting(jobToDelete);
+    try {
+      await deletePeriodicIntegrationTestCronJob(jobToDelete, namespace);
+      setCronJobs(cronJobs.filter((job) => job.metadata.name !== jobToDelete));
+      setShowSuccess('Scheduled job deleted successfully.');
+    } catch (deleteError) {
+      setError((deleteError as Error).message || 'Failed to delete job');
+    } finally {
+      setDeleting(null);
+      closeDeleteModal();
+    }
+  };
+
+  const handleOpenModal = () => {
+    if (integrationTests.length === 0) {
+      setError('No integration tests available. Please create an integration test first.');
+      return;
+    }
+    setSchedule('');
+    setSuspend(false);
+    setJobName('');
+    setJobNameError(undefined);
+    setSelectedIntegrationTest(integrationTests[0].metadata.name);
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => setIsModalOpen(false);
+
+  const generateJobName = (integrationTestName: string): string => {
+    // Create a shorter name: job-{short-test-name}-{timestamp}
+    // Limit integration test name to 20 chars to leave room for prefix and timestamp
+    // Remove any trailing dashes and replace internal dashes with hyphens
+    const shortTestName = integrationTestName
```

Comment:
i guess we need a utils for this component.
In the util, we could cover the generateJobName, create, delete and other actions.
In this way, the component could be much clean.
See the example:
components/UserAccess/UserAccessForm/UserAccessFormPage.
src/components/UserAccess/UserAccessForm/form-utils.ts